flathunter/config.py

```diff
@@ -27,6 +24,10 @@ def __init__(self, filename=None, string=None):
         self.__log__.info("Using config %s", filename)
         with open(filename) as file:
             self.config = yaml.safe_load(file)
+        self.__searchers__ = [CrawlImmobilienscout(self.config),
```

Comment:
In the case where the config is supplied as a string, this initialization fails the constructor returns before self.__searchers__ gets set. Need to replace the 'return' on line 21 with an 'else' for the file case so that both paths initialise self.__searchers__.

Also, here you pass self.config (which is the internal dictionary) rather than self (the config object). It's cleaner if the external API to the configuration is via the config object - I would like eventually for code paths not to directly access config keys.